pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -472,3 +467,36 @@ func (m *mgr) addRTAnnotation(obj runtime.Object, context string) (runtime.Objec
 
 	return obj, nil
 }
+
+func (m *mgr) updateClusterAnnotationandCondition(cluster *v3.Cluster, anno string, updateCondition bool) error {
+	sleep := 100
```

Comment:
Do you think there is a need to sleep for that long? It basically means the first attempt to update will happen only after ~2mins

Reply:
may be putting shorter sleeps after each unsuccessful attempt in the loop would be better?